src/v3/src/components/Widget/index.tsx

```diff
@@ -158,6 +166,18 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
   const initLanguage = useCallback(async () => {
     if (!Bundles.isLoaded(languageCode)) {
       await loadLanguage(widgetProps);
+      // Only pick Odyssey translation keys from translation bundle
+      setOdyTranslationOverrides({
+        [odyLanguageCode]: odysseyI18nResourceKeysList
+          .reduce((overrides: Partial<OdysseyI18nResourceKeys>,
+            key: typeof odysseyI18nResourceKeysList[number]) => {
+            const updatedOverrides = { ...overrides };
+            if (Bundles.login && Object.prototype.hasOwnProperty.call(Bundles.login, key)) {
+              updatedOverrides[key] = (Bundles.login as Partial<OdysseyI18nResourceKeys>)[key];
+            }
+            return updatedOverrides;
+          }, {}),
```

Comment:
nit: 

I know I said I usually would only create a util for something that is re-used or complex, but this, at first glance, seems like it would be beneficial to have a util function instead of having the logic here 😅 (I'm sorry). Not required though, I leave it totally up to you 🙏

Reply:
Yeah agreed that it seems like a bit too much business logic for this file. Could this fit in `languageUtils.ts`?